source/core/shape.cc

```diff
@@ -481,20 +501,222 @@ CompositeShape<dim>::static_copy() const
+      solid_volume += elem->displaced_volume(fluid_density, pcout);
     }
   return solid_volume;
 }
 
+
+template <int dim>
+double
+RBFShape<dim>::value(const Point<dim> &evaluation_point,
+                     const unsigned int /*component*/) const
+{
+  Point<dim> centered_point = this->align_and_center(evaluation_point);
+
+  // The computation and addition of the bounding box distance is necessary
+  // since the RBF nodes may not cover the whole simulation domain. In that
+  // case, it is assumed that the distance from the RBF object is approximately
+  // the same as the distance from the corresponding bounding box.
+  double bounding_box_distance = bounding_box->value(centered_point);
+  double value                 = std::max(bounding_box_distance, 0.0);
+
+  double dist, basis;
+  size_t number_of_nodes = weights.size();
+  // Algorithm inspired by Optimad Bitpit. https://github.com/optimad/bitpit
+  for (size_t i = 0; i < number_of_nodes; ++i)
+    {
+      dist  = (centered_point - nodes[i]).norm() / support_radii[i];
+      basis = evaluate_basis_function(basis_functions[i], dist);
+      value += basis * weights[i];
+    }
+  return value;
+}
+
+template <int dim>
+std::shared_ptr<Shape<dim>>
+RBFShape<dim>::static_copy() const
+{
+  std::shared_ptr<Shape<dim>> copy =
+    std::make_shared<RBFShape<dim>>(this->support_radii,
+                                    this->basis_functions,
+                                    this->weights,
+                                    this->nodes,
+                                    this->position,
+                                    this->orientation);
+  return copy;
+}
+
+template <int dim>
+double
+RBFShape<dim>::displaced_volume(const double fluid_density,
+                                std::shared_ptr<ConditionalOStream> pcout)
+{
+  if (pcout)
+    *pcout
+      << "Warning: For a RBF shape, the real volume will be lower than output."
+      << std::endl;
+  return bounding_box->displaced_volume(fluid_density);
+}
+
+template <int dim>
+double
+RBFShape<dim>::wendlandc2(const double distance) const
+{
+  return distance > 1.0 ? 0.0 :
+                          std::pow(1. - distance, 4.0) * (4.0 * distance + 1.0);
+}
+
+template <int dim>
+double
+RBFShape<dim>::linear(const double distance) const
+{
+  return distance > 1.0 ? 0.0 : (1.0 - distance);
+}
+
+template <int dim>
+double
+RBFShape<dim>::gauss90(const double distance) const
+{
+  double eps = std::pow(-1.0 * std::log(0.1), 0.5);
+  return std::exp(-1.0 * std::pow(distance * eps, 2.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::gauss95(const double distance) const
+{
+  double eps = std::pow(-1.0 * std::log(0.05), 0.5);
+  return std::exp(-1.0 * std::pow(distance * eps, 2.0));
+}
+
+template <int dim>
+double
+RBFShape<dim>::gauss99(const double distance) const
+{
+  double eps = std::pow(-1.0 * std::log(0.01), 0.5);
+  return std::exp(-1.0 * std::pow(distance * eps, 2.0));
+}
+
+template <int dim>
+double
```

Comment:
Why not implement those in the .h and inline them? It would be really really faster.